magefiles/rulesengine/repos/infra_deployments.go

```diff
@@ -1,17 +1,158 @@
 package repos
 
-import "github.com/konflux-ci/e2e-tests/magefiles/rulesengine"
-
-// Example rule for infra-deployments using anonymous functions embedded in the rule
-var InfraDeploymentsTestRulesCatalog = rulesengine.RuleCatalog{
-	rulesengine.Rule{Name: "Infra Deployments Default Test Execution",
-		Description: "Run the default test suites which include the demo and components suites.",
-		Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
-
-			return rctx.RepoName == "infra-deployments"
-		}),
-		Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
-			rctx.LabelFilter = "e2e-demo,konflux-demo,spi-suite,remote-secret,integration-service,ec,build-templates,multi-platform"
-			return ExecuteTestAction(rctx)
-		})}},
+import (
+	"fmt"
+	"strings"
+
+	"github.com/konflux-ci/e2e-tests/magefiles/rulesengine"
+)
+
+// Default Rule of repo infra-deployments running konflux-demo suite.
+var InfraDeploymentsDefaultTestRules = rulesengine.Rule{Name: "Infra Deployments Default Test Execution",
+	Description: "Run the Konflux-demo suite tests when an Infra-deployments PR includes changes to files outside of the specified components.",
+	Condition: rulesengine.None{
+		&InfraDeploymentsIntegrationTestFileChangeRule,
+		&InfraDeploymentsImageControllerTestFileChangeRule,
+		&InfraDeploymentsMultiPlatformTestFileChangeRule,
+		&InfraDeploymentsBuildTemplatesTestFileChangeRule,
+		&InfraDeploymentsBuildServiceTestFileChangeRule,
+		&InfraDeploymentsReleaseServiceTestFileChangeRule,
+		&InfraDeploymentsEnterpriseControllerTestFileChangeRule,
+		&InfraDeploymentsJVMTestFileChangeRule,
+		rulesengine.ConditionFunc(CheckNoFilesChanged)},
+
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(ExecuteInfraDeploymentsDefaultTestAction)}}
+
+// ExecuteInfraDeploymentsDefaultTestAction excutes all the e2e-tests and component suites
+func ExecuteInfraDeploymentsDefaultTestAction(rctx *rulesengine.RuleCtx) error {
+	rctx.LabelFilter = "konflux-demo"
```

Comment:
the label filter name for the demo test is [`konflux`](https://github.com/konflux-ci/e2e-tests/blob/602a159f8ff8b26993209baad53101bd40ace95c/tests/konflux-demo/const.go#L27)